desk/sur/forums.hoon

```diff
@@ -239,142 +245,142 @@
       :-  metadata.rock
       =<  +
       %+  ~(q db database.rock)
-        %forums  
+        %forums
       :*  %update  post-table  [%s %post-id [%& %eq post-id.act]]
           :~  :-  %votes
               |=  votes=value
               ^-  value
               ?>  ?=(%m -.votes)
-              :-  %m  
+              :-  %m
               ?.  (~(has by p.votes) src.bowl)
                 (~(put by p.votes) src.bowl dir.act)
               (~(del by p.votes) src.bowl)
       ==  ==
     ::
         %delete-post
-      ::  TODO: 
+      ::  TODO:
       ::  1. Scry groups to obtain permissions if channel is not null
       ::  ?~  channel.metadata.rock
       ::    %&
       ::  :: JOIE:  Alias the block below using =* instead of =/
-      ::  =/  allow-groups=? 
-      ::  ?^  channel 
+      ::  =/  allow-groups=?
+      ::  ?^  channel
       ::    <<scry groups for permissions>>
       ::    ?:  <<allowed in groups>>  %&  %|
       ::  %&
-      ::  
-      =/  author=@p  author:(get-post database.rock post-table id.act)
+      ::
+      ::  TODO: Extend this function to remove all orphaned posts (i.e.
+      ::  the entire subtree of comments below a post).
+      =/  act-post=post  (get-post database.rock post-table post-id.act)
+      =/  [[@da author=@p @t] edits]  (pop:om-hist p.history.act-post)
       ::  2. Check if src.bowl is author OR has the appropriate permissions
       ::  ?>  |&  =(author src.bowl)  allow-groups
-      ?>  =(author src.bowl)
+      ?.  =(author src.bowl)
+        ~|('%forums: deletion permissions denied' !!)
+      ::
+      ::  remove child references to this post in parents (post, thread)
+      ::
+      =?  database.rock  !=(parent-id.act-post 0)
+        =<  +
+        %+  ~(q db database.rock)  %forums
+        :*  %update  post-table  [%s %post-id [%& %eq parent-id.act-post]]
+        :~  :-  %child-ids
+            |=  child-ids=value
+            ^-  value
+            ?>  ?=(%s -.child-ids)
+            [%s (~(del in p.child-ids) post-id.act)]
+        ==  ==
```

Comment:
```suggestion
        :*  %update  post-table  [%s %post-id [%& %eq parent-id.act-post]]
             :~  :-  %child-ids
                  |=  child-ids=value
                  ^-  value
                  ?>  ?=(%s -.child-ids)
                  [%s (~(del in p.child-ids) post-id.act)]
        ==  ==
```

Reply:
Fixed (though, why is this indented twice instead of once?)